.github/workflows/publish-images.yml

```diff
@@ -0,0 +1,58 @@
+name: publish-images
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
+
+jobs:
+  publish-image:
+    strategy:
+      matrix:
+        component:
+          - tempo
+          - tempo-query
+          - tempo-vulture
+    runs-on:
+      - ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+
+      - id: prepare
+        name: Prepare
+        run: |-
+          HUB=docker.io/${{ github.repository_owner }}
+          GITHUB_SHA=${{ github.sha }}
+
+          IMAGES=${HUB}/${{ matrix.component }}:latest,${HUB}/${{ matrix.component }}:${GITHUB_SHA::8}
+
+          if [[ ${{ github.ref }} == "refs/tags/v"* ]]; then
+            GITHUB_REF=${{ github.ref }}
+            IMAGES=${HUB}/${{ matrix.component }}:${GITHUB_REF/refs\/tags\/v/}
+          fi
+
+          echo ::set-output name=image::${IMAGES//$','/'%0A'}
```

Comment:
to pass multi line image tag list with `\n`

https://github.community/t/set-output-truncates-multiline-strings/16852